lib/enumerable_2.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_count(item = nil)
+    count = 0
+    if block_given?
+      my_each { |i| count += 1 if yield(i) }
+    elsif !item.nil?
+      my_each { |i| count += 1 if i == item }
+    else
+      return size
+    end
+    count
+  end
+
+  def my_map(proc = nil)
+    arr = []
+    if proc
+      my_each { |i| arr.push(proc.call(i)) }
+    elsif block_given?
+      my_each { |i| arr.push(yield(i)) }
+    else
+      return my_each
+    end
+    arr
+  end
+
+  def my_inject(*args)
+    values = get_initial_symbol(args)
+    initial = values[0]
+    sym = values[1]
+
+    if block_given?
+      my_each_with_index do |item, index|
+        initial = args.empty? && index.zero? ? initial : yield(initial, item)
+      end
+    else
+      my_each_with_index do |item, index|
+        initial = args.size == 1 && index.zero? ? initial : initial.send(sym, item)
+      end
+    end
+    initial
+  end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] combines all elements of enum by applying a binary operation, specified by a block
- [ ] when a symbol is specified combines each element of the collection by applying the symbol as a named method

Reply:
https://github.com/fivan18/enumerables/blob/77f2de9dc296c28b1e9e7763667b155d0df44fcb/spec/enumerable_spec.rb#L142-L147